velox/functions/prestosql/StringFunctions.h

```diff
@@ -267,4 +267,77 @@ struct StrLPosFunction : public StrPosFunctionBase<T, true> {};
 template <typename T>
 struct StrRPosFunction : public StrPosFunctionBase<T, false> {};
 
+template <typename T>
+struct LevenshteinDistanceFunction {
+  VELOX_DEFINE_FUNCTION_TYPES(T);
+
+  FOLLY_ALWAYS_INLINE void call(
+      out_type<int64_t>& result,
+      const arg_type<Varchar>& left,
+      const arg_type<Varchar>& right) {
+    doCall<false>(result, left, right);
+  }
+
+  FOLLY_ALWAYS_INLINE void callAscii(
+      out_type<int64_t>& result,
+      const arg_type<Varchar>& left,
+      const arg_type<Varchar>& right) {
+    doCall<true>(result, left, right);
+  }
+
+  template <bool isAscii>
+  FOLLY_ALWAYS_INLINE void doCall(
+      out_type<int64_t>& result,
+      const arg_type<Varchar>& left,
+      const arg_type<Varchar>& right) {
+    std::vector<int32_t> leftCodePoints =
```

Comment:
Let's refactor a bit:

1- Move stringToCodePoint calls into the caller and change the signature of doCall to take 2 vectors of code points.
2- Add template parameter TCodePoint to doCall to indicate the type of the code point. It will be int32_t for non-Ascii case and int8_t for Ascii case. This way you won't need to create a vector of int32_t for Ascii case and will be passing the input strings directly (no copy).